bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java

```diff
@@ -574,7 +574,9 @@ public void removeRule(TriggerTypes type, Rule rule) {
                 }
                 break;
             case TRIGGER:
-                triggerEventTriggeredRules.remove(rule);
+                for (Set<Rule> rules : triggerEventTriggeredRules.values()) {
+                    rules.remove(rule);
```

Comment:
So I _could_ remove something from the map I iterate over, if I wanted. But I don't, as @maggu2810 correctly pointed out.